akka-projection-grpc/src/main/scala/akka/projection/grpc/replication/internal/ReplicationImpl.scala

```diff
@@ -164,40 +164,74 @@ private[akka] object ReplicationImpl {
+                      case Some(replicatedEventMetadata: ReplicatedEventMetadata) =>
+                        // skipping events originating from other replicas is handled by filtering but for good measure
+                        require(replicatedEventMetadata.originReplica == remoteReplica.replicaId)
+
+                        val replicationId = ReplicationId.fromString(envelope.persistenceId)
+                        val destinationReplicaId = replicationId.withReplica(settings.selfReplicaId)
+                        val entityRef =
+                          entityRefFactory(destinationReplicaId.entityId).asInstanceOf[EntityRef[PublishedEvent]]
+                        if (log.isDebugEnabled) {
+                          log.debugN(
+                            "[{}], Forwarding event originating on dc [{}] to [{}] (version: [{}]): [{}]",
+                            system.name,
+                            replicatedEventMetadata.originReplica,
+                            destinationReplicaId.persistenceId.id,
+                            replicatedEventMetadata.version,
+                            event)
+                        }
+                        val askResult = entityRef.ask[Done](replyTo =>
+                          PublishedEventImpl(
+                            replicationId.persistenceId,
+                            replicatedEventMetadata.originSequenceNr,
+                            event,
+                            envelope.timestamp,
+                            Some(new ReplicatedPublishedEventMetaData(
+                              replicatedEventMetadata.originReplica,
+                              replicatedEventMetadata.version)),
+                            Some(replyTo)))
+                        askResult.failed.foreach(error =>
+                          log.warn(s"Failing replication stream from [${remoteReplica.replicaId.id}]", error))
+                        askResult
+
+                      case unexpected =>
+                        throw new IllegalArgumentException(
+                          s"Got unexpected type of event envelope metadata: ${unexpected.getClass} (pid [${envelope.persistenceId}], seq_nr [${envelope.sequenceNr}]" +
+                          ", is the remote entity really a Replicated Event Sourced Entity?")
+                    }
+
+                  case None =>
+                    // FIXME the event is not filtered because surrounding if-else, but we do not propagate the source == "BT"
+                    //       flag, so we can't compare against that, should we?
+                    // (deduplication happens in the receiving entity)
+                    log.debug(
+                      "Loading backtracking event (pid: [{}], seq_nr: [{}])",
+                      envelope.persistenceId,
+                      envelope.sequenceNr)
+                    eventsBySlicesQuery
+                      .loadEnvelope[AnyRef](envelope.persistenceId, envelope.sequenceNr)
```

Comment:
I think this is wrong. `loadEnvelope` is done from the R2dbcProjectionImpl here https://github.com/akka/akka-persistence-r2dbc/blob/main/projection/src/main/scala/akka/projection/r2dbc/internal/R2dbcProjectionImpl.scala#L330
and that is only done after `isAccepted`, i.e. only when needed

Reply:
Ok, it was surprising that it started to crash there, but maybe that was just because I didn't set the bt flat correctly.